### ISSUE TITLE
Not bundling extra entries

### ISSUE THREAD

**Issue by nitedani** [nitedani/vite-plugin-standalone#1]
Quoting @simplecommerce
```
It seems to compile the code and add it to the bundle but it doesn't seem to bundle the imports that the worker code has inside.

For example, in my code I have:

import WebSocket from "ws";
import { createClient } from "graphql-ws";
import { isEmpty } from "lodash-es";
And in the output code I see

import o from"ws";import{createClient as r}from"graphql-ws";import{isEmpty as n}from"lodash-es";
Am I missing something?
```

**Comment by nitedani** [comment]
I updated the plugin recently, please check the latest version 1.0.6.
The interface has changed: `standalone({ entry: ["./src/worker.js"] })`

I plan to make the `entry` option more explicit. Currently it includes the rollup entries by default, which can cause issues.

**Comment by simplecommerce** [comment]
> entry

I just tested it and I get this issue now:

```
error during build:
Error: Build failed with 3 errors:
dist/server/entries/renderer_error.mjs:5228:20: ERROR: The symbol "dirname987" has already been declared
dist/server/entries/renderer_error.mjs:5229:26: ERROR: The symbol "fileURLToPath987" has already been declared
dist/server/entries/renderer_error.mjs:5230:26: ERROR: The symbol "createRequire987" has already been declared
    at failureErrorWithLog (/root/v3/core-vite-ts/backend/node_modules/esbuild/lib/main.js:1651:15)
    at /root/v3/core-vite-ts/backend/node_modules/esbuild/lib/main.js:1059:25
    at runOnEndCallbacks (/root/v3/core-vite-ts/backend/node_modules/esbuild/lib/main.js:1486:45)
    at buildResponseToResult (/root/v3/core-vite-ts/backend/node_modules/esbuild/lib/main.js:1057:7)
    at /root/v3/core-vite-ts/backend/node_modules/esbuild/lib/main.js:1086:16
    at responseCallbacks.<computed> (/root/v3/core-vite-ts/backend/node_modules/esbuild/lib/main.js:704:9)
    at handleIncomingPacket (/root/v3/core-vite-ts/backend/node_modules/esbuild/lib/main.js:764:9)
    at Socket.readFromStdout (/root/v3/core-vite-ts/backend/node_modules/esbuild/lib/main.js:680:7)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:315:12)
```

And the code in my worker file still contains the imports without bundling them.
I have no idea what is causing it. I'll try to see if I can create a sample project so you could see it also.

**UPDATE**

I haven't found the issue and I haven't been able to reproduce it yet in a small test.
But what seems to be happening is the 3 following imports are found twice in the `renderer_error.mjs` file for some reason:

```
import { dirname as dirname987 } from 'path';
import { fileURLToPath as fileURLToPath987 } from 'url';
import { createRequire as createRequire987 } from 'module';
```

**Comment by nitedani** [comment]
I see. I think I know the cause, I'll push an update tonight.

**Comment by nitedani** [comment]
Released 1.0.7
New usage:
```ts
    standalone(),
```
or
```ts
    standalone({
      entry: {
        worker: "./server/worker.js",
      },
    }),
```
or
```ts
    standalone({
      entry: {
        index: "./server/index.ts",
        worker: "./server/worker.js",
      },
    }),
```

If `entry.index` is set, it won't try to auto-detect the index entry.

**Comment by simplecommerce** [comment]
> Released 1.0.7 New usage:
> 
> ```ts
>     standalone(),
> ```
> 
> or
> 
> ```ts
>     standalone({
>       entry: {
>         worker: "./server/worker.js",
>       },
>     }),
> ```
> 
> or
> 
> ```ts
>     standalone({
>       entry: {
>         index: "./server/index.ts",
>         worker: "./server/worker.js",
>       },
>     }),
> ```
> 
> If `entry.index` is set, it won't try to auto-detect the index entry.

It worked this time, no more build error.

I do see these types of warnings but I assume its normal because it doesn't seem to break:

```
▲ [WARNING] Ignoring this import because "node_modules/react-router-dom/server.mjs" was marked as having no side effects [ignored-bare-import]

    dist/server/chunks/chunk-i3XHNZHX.js:1:1350:
      1 │ ...vike/server";import"@emotion/server/create-instance";import"react-router-dom/server.mjs";import"ua-parser-js";import"react-i18next";import"graphql...
        ╵                                                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  "sideEffects" is false in the enclosing "package.json" file:

    node_modules/react-router-dom/package.json:20:2:
      20 │   "sideEffects": false,
         ╵   ~~~~~~~~~~~~~

▲ [WARNING] Ignoring this import because "node_modules/graphql-tag/main.js" was marked as having no side effects [ignored-bare-import]

    dist/server/chunks/chunk-i3XHNZHX.js:1:1429:
      1 │ ...r-dom/server.mjs";import"ua-parser-js";import"react-i18next";import"graphql-tag";import"tslib";import"@wry/equality";import"./chunk-AuiZ6p8K.js";i...
        ╵                                                                       ~~~~~~~~~~~~~

  "sideEffects" is false in the enclosing "package.json" file:

    node_modules/graphql-tag/package.json:9:2:
      9 │   "sideEffects": false,
        ╵   ~~~~~~~~~~~~~

▲ [WARNING] Ignoring this import because "node_modules/tslib/tslib.es6.mjs" was marked as having no side effects [ignored-bare-import]

    dist/server/chunks/chunk-i3XHNZHX.js:1:1449:
      1 │ ...;import"ua-parser-js";import"react-i18next";import"graphql-tag";import"tslib";import"@wry/equality";import"./chunk-AuiZ6p8K.js";import"ts-invarian...
        ╵                                                                          ~~~~~~~

  "sideEffects" is false in the enclosing "package.json" file:

    node_modules/tslib/package.json:28:4:
      28 │     "sideEffects": false,
         ╵     ~~~~~~~~~~~~~
```

And I just ran my test project and it works fine.

This is awesome!

**Comment by simplecommerce** [comment]
I don't know if its related, for some reason when I run my app using the dev on Node v18.17.1.
It won't load at all.
But if I downgrade to v16.20.2 and reinstall my node_modules and try again, it works no problem.

I don't get any errors in the server side, it seems to want to load the site and gives me a 200 response but all I see is: ERR_HTTP2_PROTOCOL_ERROR in the browser.

I don't know if it has anything to do with the plugin or a combination of this with vavite, but just thought I'd mention it.

**UPDATE**

Sorry about that, its not your plugin.

**Comment by nitedani** [comment]
The warnings are fixed in 1.0.8
